### PR TITLE
Add search cutoffs, null handing, and refresh API reference

### PR DESCRIPTION
**Pull request by pwoznic**
Added information about search cutoffs, handling null values, and regenerated the API reference.

### CODE REVIEW

File: www/docs/api-reference/search-apis/reranking.md [hunk 1]
```diff
@@ -18,13 +18,26 @@ of query results to achieve a more precise ranking. It employs a machine
...
+  provides more accurate neural ranking than the initial Boomerang retrieval. 
+  While computationally more expensive, it offers improved text scoring across 
+  a wide range of languages, making it suitable for diverse content..
+* [**Maximal Marginal Relevance (MMR) Reranker**](/docs/learn/mmr-reranker) (`type=mmr`) 
+  for diversifying results while maintaining relevance.
+* [**User Defined Function Reranker**](/docs/learn/user-defined-function-reranker) (`type=userfn`) for 
+  custom scoring based on metadata.
+
+
+### Chain Reranker
```
Comment (husseinVectara) [inline www/docs/api-reference/search-apis/reranking.md:35]: Lets call this Chain Reranking

File: www/docs/api-reference/search-apis/search.md [hunk 2]
```diff
@@ -189,10 +189,13 @@ more rarely, as a response.
 ### Reranking Configuration
 
 The `reranker` object enables the reranking of query results, to further 
-increase relevance in certain scenarios. For more details, see [Rerank Search Results](/docs/api-reference/search-apis/reranking):
+increase relevance in certain scenarios. The `limit` property of the `reranker` 
+object allows you to have more granular control over the number of results 
+returned after rerankingFor more details, see [Rerank Search Results](/docs/api-reference/search-apis/reranking):
```
Comment (husseinVectara) [inline www/docs/api-reference/search-apis/search.md:194]: missing a space hahaha

File: www/docs/api-reference/search-apis/reranking.md [hunk 3]
```diff
@@ -59,8 +72,30 @@ this simplified example intentionally omits several parameter values.
...
+
+The `limit` property of the `reranker` object allows you to have more 
+granular control over the number of results returned after reranking. This 
+limit is applied per each reranking stage, such as if you use chain reranking, 
+and this limit affects the output and not the input to the reranker. 
+When you apply this limit to the Multilingual reranker, Maximal Marginal Relevance 
+(MMR) Reranker, or User Defined Function reranker, it performs the following 
+steps:
+
+1. Reranks all input results according to its algorithm.
```
Comment (husseinVectara) [inline www/docs/api-reference/search-apis/reranking.md:85]: Lets say 'Reranks all input results based on the selected Reranker.'.

File: www/docs/api-reference/search-apis/reranking.md [hunk 4]
```diff
@@ -23,9 +23,10 @@ accurate results.
 Vectara currently provides the following rerankers: 
 
 * [**Multilingual Reranker v1**](/docs/learn/vectara-multi-lingual-reranker) (`type=customer_specific` and `reranker_id=rnk_272725719`)
```
Comment (husseinVectara) [inline www/docs/api-reference/search-apis/reranking.md:25]: reranker id is deprecated, lets write here reranker_name = Rerank_Multilingual_v1

File: www/docs/api-reference/search-apis/reranking.md [hunk 5]
```diff
@@ -68,19 +69,54 @@ this simplified example intentionally omits several parameter values.
...
 and this limit affects the output and not the input to the reranker. 
-When you apply this limit to the Multilingual reranker, Maximal Marginal Relevance 
-(MMR) Reranker, or User Defined Function reranker, it performs the following 
-steps:
+When you apply this limit to the Multilingual reranker (Slingshot), Maximal 
+Marginal Relevance (MMR) reranker, or User Defined Function reranker, it 
+performs the following steps:
 
 1. Reranks all input results based on the selected reranker.
 2. Eliminates results that return null scores.
```
Comment (husseinVectara) [inline www/docs/api-reference/search-apis/reranking.md:122]: lets add: Applies score cutoff if specified after here, and standardize this list between the limits and the cutoff's description.

File: www/docs/api-reference/search-apis/reranking.md [hunk 6]
```diff
@@ -98,4 +134,42 @@ content.
...
+    "rerankers": [
+      {
+        "type": "userfn",
+        "user_function": "get('$.document_metadata.category') === 'blog' ? get('$.score') : null",
+        "cutoff": 0.5,
+        "limit": 10
+      },
+      {
+        "type": "customer_specific",
+        "reranker_id": "rnk_272725719",
```
Comment (husseinVectara) [inline www/docs/api-reference/search-apis/reranking.md:163]: "reranker_name" = "Rerank_Multilingual_v1"

File: www/docs/api-reference/search-apis/reranking.md [hunk 7]
```diff
@@ -98,4 +134,42 @@ content.
...
+
+```json
+{
+  "reranker": {
+    "type": "chain",
+    "rerankers": [
+      {
+        "type": "userfn",
+        "user_function": "get('$.document_metadata.category') === 'blog' ? get('$.score') : null",
+        "cutoff": 0.5,
```
Comment (husseinVectara) [inline www/docs/api-reference/search-apis/reranking.md:158]: lets remove cutoff from here as its redundant with limit in the next reranker

File: www/docs/api-reference/search-apis/reranking.md [hunk 8]
```diff
@@ -98,4 +134,42 @@ content.
...
+        "limit": 10
+      },
+      {
+        "type": "customer_specific",
+        "reranker_id": "rnk_272725719",
+        "cutoff": 0.5,
+        "limit": 10
```
Comment (husseinVectara) [inline www/docs/api-reference/search-apis/reranking.md:165]: lets change this to limit 3 instead of 10 so its not redundant

File: www/docs/api-reference/search-apis/reranking.md [hunk 9]
```diff
@@ -98,4 +134,42 @@ content.
...
+      }
+    ]
+  }
+}
+```
+This filters out non-blog content with the UDF reranker and then uses the 
+Vectara Multilingual reranker on the remaining results which both removes
```
Comment (husseinVectara) [inline www/docs/api-reference/search-apis/reranking.md:172]: lets highlight that the limit of 10 limits on udf limits the output being sent to Slingshot for efficiency and speed.

File: www/docs/learn/vectara-multi-lingual-reranker.md [hunk 10]
```diff
@@ -9,9 +9,30 @@ import TabItem from '@theme/TabItem';
...
+
+While more computationally expensive and introducing some additional latency, 
+the multilingual reranker improves neural ranking beyond Boomerang’s initial 
+selection by providing more precise text scoring. Think of the Slingshot 
+reranker as a "better Boomerang" for refining results, with the multilingual 
+capability serving primarily as a differentiator from other rerankers in the 
+market, which are often English-only.
+
+Using this reranker requires both the `type` and `reranker_id` in the 
 `reranker` object. Set the `type` as `customer_reranker` and the `reranker_id` as `rnk_272725719`.
```
Comment (husseinVectara) [inline www/docs/learn/vectara-multi-lingual-reranker.md:36]: lets use reranker_name here

File: www/docs/api-reference/search-apis/reranking.md [hunk 11]
```diff
@@ -55,12 +68,120 @@ this simplified example intentionally omits several parameter values.
   "enable_factual_consistency_score": true
 }
 ```
-
+:::tip
```
Comment (ofermend) [inline www/docs/api-reference/search-apis/reranking.md:71]: Small nit: when I read something like "you can also enable reranking in the Vectara console..." it may be confusing in the sense that I ask myself: "if I enable it in console, does that affect my application?". Of course the answer is not, but I wonder if we can rephrase it to reiterate the message that console is for exploration, experimentation, etc with the API. but it ultimately of course the API call itself is what makes it happen.

File: www/docs/api-reference/search-apis/reranking.md [hunk 12]
```diff
@@ -55,12 +68,120 @@ this simplified example intentionally omits several parameter values.
...
+granular control over the number of results returned after reranking. This 
+limit is applied per each reranking stage, such as if you use chain reranking, 
+and this limit affects the output and not the input to the reranker. 
+When you apply this limit to the Multilingual reranker (Slingshot), Maximal 
+Marginal Relevance (MMR) reranker, or User Defined Function reranker, it 
+performs the following steps:
+
+1. Applies score cutoff, if specified.
+2. Reranks all input results based on the selected reranker.
+3. Eliminates results that return null scores.
```
Comment (ofermend) [inline www/docs/api-reference/search-apis/reranking.md:129]: What are the cases where this might be happening? How would a developer know if any results return NULL scores?

File: www/docs/api-reference/search-apis/reranking.md [hunk 13]
```diff
@@ -55,12 +68,120 @@ this simplified example intentionally omits several parameter values.
...
+query results. 
+
+```json
+{
+  "reranker": {
+    "type": "chain",
+    "rerankers": [
+      {
+        "type": "userfn",
+        "user_function": "get('$.document_metadata.category') === 'blog' ? get('$.score') : null",
```
Comment (ofermend) [inline www/docs/api-reference/search-apis/reranking.md:170]: So maybe this is related to my question about NULL above - is this the only use-case, where my UDF decided to return NULL?

File: www/docs/api-reference/search-apis/reranking.md (hunk 12 above)
Comment (husseinVectara) [reply www/docs/api-reference/search-apis/reranking.md:129]: I think I should have explained this better, null scores are only returned deliberately by user functions. So if you want to eliminate some search results, have a udf return null (say apply a metadata filter)

File: www/docs/api-reference/search-apis/reranking.md (hunk 13 above)
Comment (husseinVectara) [reply www/docs/api-reference/search-apis/reranking.md:170]: yes this is the only case where scores will be eliminated (aside from cutoffs and limits)

**Comment by pwoznic** [comment]
There were several conflicts so trying to sort it out. May take more commits.

**Review by eskibars — APPROVED** [review]

File: www/docs/api-reference/search-apis/reranking.md (hunk 13 above)
Comment (husseinVectara) [reply www/docs/api-reference/search-apis/reranking.md:163]: Please correct this to:
"if (get('$.document_metadata.category') == 'blog')  get('$.score') else null"
